src/Adapter/Order/CommandHandler/CancelOrderProductHandler.php

```diff
@@ -31,35 +31,51 @@
 use Customization;
 use Hook;
 use Order;
-use OrderCarrier;
 use OrderDetail;
-use PrestaShop\PrestaShop\Adapter\Order\Refund\OrderProductRemover;
+use OrderHistory;
+use PrestaShop\PrestaShop\Adapter\Order\OrderProductQuantityUpdater;
 use PrestaShop\PrestaShop\Core\Domain\Order\Command\CancelOrderProductCommand;
 use PrestaShop\PrestaShop\Core\Domain\Order\CommandHandler\CancelOrderProductHandlerInterface;
 use PrestaShop\PrestaShop\Core\Domain\Order\Exception\InvalidCancelProductException;
 use PrestaShop\PrestaShop\Core\Domain\Order\Exception\InvalidOrderStateException;
-use StockAvailable;
-use Validate;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Translation\TranslatorInterface;
 
 /**
  * @internal
  */
 final class CancelOrderProductHandler extends AbstractOrderCommandHandler implements CancelOrderProductHandlerInterface
 {
     /**
-     * @var OrderProductRemover
+     * @var OrderProductQuantityUpdater
      */
-    private $orderProductRemover;
+    private $orderProductQuantityUpdater;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
 
     /**
      * CancelOrderProductHandler constructor.
      *
-     * @param OrderProductRemover $orderProductRemover
+     * @param OrderProductQuantityUpdater $orderProductQuantityUpdater
+     * @param LoggerInterface $logger
+     * @param TranslatorInterface $translator
      */
     public function __construct(
-        OrderProductRemover $orderProductRemover
+        OrderProductQuantityUpdater $orderProductQuantityUpdater,
+        LoggerInterface $logger,
```

Comment:
It's a BC with you add new parameters to the construct

Reply:
But this handler doesn't exist in previous versions :D

Reply:
In the adapter namescape BC are usually accepted although they need to be documented
But in this case indeed it's a new class for 177 so no BC here

Reply:
Ok. Bad one for me. Sorry.